### ISSUE TITLE
Web UI display error after adding annotations via REST API.

### ISSUE THREAD

**Issue by mbspng** [heartexlabs/label-studio#3581]
What does this mean?

```
Error: [mobx-state-tree] No matching type for union ({ pid: string?; score: (number | null?); readonly: boolean?; hidden: boolean?; parentID: (string | null?); fromSuggestion: boolean?; dynamic: boolean?; locked: boolean?; origin: ("prediction" | "prediction-changed" | "manual")?; meta: ?; normInput: (string | null?); id: identifier?; ouid: number?; results: { id: identifier?; score: (number | null?); from_name: reference((AudioModel | AudioModel | PolygonRegionModel | BrushRegionModel | KeyPointRegionModel | RectRegionModel | EllipseRegionModel | ImageModel | LabelModel | LabelsModel | ParagraphLabelsModel | TextAreaRegionModel | TextAreaModel | ChoiceModel | ChoicesModel | RatingModel | ParagraphsModel | RichTextModel | RichTextModel | TableModel | ChannelModel | TimeSeriesModel | PagedViewModel | TimeSeriesRegionModel | RelationsModel | VideoModel | BrushModel | BrushLabelsModel | HyperTextLabelsModel | KeyPointModel | KeyPointLabelsModel | ListModel | DateTimeM
```

![image](https://user-images.githubusercontent.com/10274874/213731490-a886cb3c-1143-452e-9ee6-794486ae3179.png)

I get this when I try to display images in the labeling UI after using the REST endpoint to create an annotation.

I am using the ML Backend to first query the backend for predictions, and then I add these predictions as annotations because otherwise they do not end up in the database, apparently.

```python
def predict_for_all_tasks(client: Client, project: Project):

    tasks = take(10, project.get_tasks())
    logger.info(f"Predicting for {pluralize(len(tasks), 'task')}...")

    response = requests.post(
        headers={"Authorization": f"Token {client.api_key}"},
        url="http://localhost:9090/predict",
        json={"tasks": tasks},
    )
    logger.debug(f"Response for prediction:\n {json.dumps(response.json(), indent=2)}")
    response.raise_for_status()
    logger.success(f"Prediction successful.")
    return response.json()["results"]


def create_annotations_from_predictions(client: Client, predictions: List[dict]):
    """
    https://labelstud.io/api#operation/api_tasks_annotations_create

    curl -X POST -H 'Content-type: application/json' https://localhost:8080/api/tasks/{id}/annotations -H 'Authorization
    : Token abc123' --data '{...}'
    """

    logger.info(f"Creating annotations for {pluralize(len(predictions), 'task')}...")

    for prediction in predictions:
        task_id = prediction["result"][0]["task_id"]  # TODO: This is wonky
        logger.info(f"Creating annotation for task {task_id}...")

        payload = {
            "completed_by": 1,  # this has to be an existing user id, otherwise the request fails
            "unique_id": uuid(),
            "result": prediction["result"],
            "was_cancelled": True,
            "ground_truth": True,
            "last_action": "prediction",
            "task": task_id,
            "project": None,
            "updated_by": None,
            "parent_prediction": None,
            "parent_annotation": None,
            "last_created_by": None,
        }

        response = requests.post(
            headers={"Authorization": f"Token {client.api_key}"},
            url=client.get_url(f"/api/tasks/{task_id}/annotations"),
            json=payload,
        )
        logger.debug(f"Response for annotation creation:\n {json.dumps(response.json(), indent=2)}")
        response.raise_for_status()
        logger.success(f"Annotation successful.")
```

Calling code:

```python
client = initialize_label_studio()
project = initialize_project(client)
...
predictions = predict_for_all_tasks(client, project)
create_annotations_from_predictions(client, predictions)
```

```
label-studio                   1.7.0
label-studio-converter         0.0.48rc0
label-studio-ml                1.0.9
label-studio-sdk               0.0.16
label-studio-tools             0.0.1
```

**Comment by mbspng** [comment]
So for what it's worth: Doing it like this works:

```python
predictions = predict_for_all_tasks(client, project)

for prediction in predictions:
    result = prediction["result"]
    task_id = result[0]["task_id"]
    project.create_prediction(task_id, result)
```

This way the predictions end up in the UI and the data (images for me) still get displayed correctly.

Working with this SDK is not a fun experience!!

**Comment by mbspng** [comment]
Okay, why on earth does the ML backend get triggered again when I press "Label All Tasks" in the GUI? The tasks have already been predicted for! That is why I am doing all of this, so that the predictions get produced beforehand! So that one does not have to wait for the inference to run for the next task every time on submits a task!

**Comment by makseq** [comment]
1. 
> What does this mean?

Your `prediction["result"]` mismatches with your labeling config. Most likely you use incorrectly `from_name` field. 

2.
> why on earth does the ML backend get triggered again when I press "Label All Tasks" in the GUI?

Most likely this toggle is enabled in your setup. ML backend is triggered because of active learning strategy: by design when you click submit, you retrain your model. Then ML backend retrieve the newest predictions from the newest model. 

![image](https://user-images.githubusercontent.com/501780/213778209-1a106bba-6d92-45ae-8fb2-68e808c28d61.png)

**Comment by mbspng** [comment]
@makseq No, the toggle is off when I go to the project settings. Also, the `from_name` is not differing. I looked at the diff between a prediction produces by triggering the process from the GUI and a prediction produced via the REST API.

I cannot spot a difference that seems relevant in the two dictionaries.


not working:
```
{
  "annotations": [
    {
      "completed_by": {
        "avatar": null,
        "email": "ja.geh@doch.weg",
        "first_name": "",
        "id": 1,
        "initials": "ja",
        "last_name": ""
      },
      "created_ago": "0 minutes",
      "created_at": "2023-01-20T15:28:53.843705Z",
      "created_username": " ja.geh@doch.weg, 1",
      "ground_truth": true,
      "id": 126,
      "lead_time": null,
      "parent_annotation": null,
      "parent_prediction": null,
      "project": 170,
      "result": [
        {
          "from_name": "tag",
          "id": "CCEh5geMrzbKa3TYdCrNyM",
          "image_rotation": 0,
          "origin": "inference",
          "original_height": 528,
          "original_width": 624,
          "task_id": 1235,
          "to_name": "image",
          "type": "brushlabels",
          "value": {
            "brushlabels": [
              "label1"
            ],
            "format": "rle",
            "rle": []
          }
        }
      ],
      "task": 1235,
      "updated_at": "2023-01-20T15:28:53.843725Z",
      "was_cancelled": true
    }
  ],
  "data": {
    "image": "/data/local-files/?d=/images/7b3a6559b521f7d27bafde64b6e67a1a.jpeg"
  },
  "id": 1235,
  "predictions": []
}
```

working:
```
{
  "annotations": [
    {
      "completed_by": {
        "avatar": null,
        "email": "dev-annotator-1@dev.dev",
        "first_name": "",
        "id": 3,
        "initials": "de",
        "last_name": ""
      },
      "created_ago": "0 minutes",
      "created_at": "2023-01-20T15:30:57.891778Z",
      "created_username": " dev-annotator-1@dev.dev, 3",
      "ground_truth": false,
      "id": 136,
      "lead_time": 56.564,
      "parent_annotation": null,
      "parent_prediction": 1426,
      "project": 170,
      "result": [
        {
          "from_name": "tag",
          "id": "bdkLi7VCdYMhoCxnaDBkJx",
          "image_rotation": 0,
          "origin": "prediction",
          "original_height": 528,
          "original_width": 624,
          "to_name": "image",
          "type": "brushlabels",
          "value": {
            "brushlabels": [
              "label1"
            ],
            "format": "rle",
            "rle": []
          }
        }
      ],
      "task": 1245,
      "updated_at": "2023-01-20T15:30:57.891797Z",
      "was_cancelled": false
    }
  ],
  "data": {
    "image": "/data/local-files/?d=/images/4c588c8647e45a1670feb0cd16db08fe.jpeg"
  },
  "id": 1245,
  "predictions": []
}
```

**Comment by mbspng** [comment]
> ML backend is triggered because of active learning strategy: by design when you click submit, you retrain your model. Then ML backend retrieve the newest predictions from the newest model.

@makseq Okay. Why? That makes no sense to me. The user presses "submit" to confirm an annotation. Why would the design be to re-train the model when a single new annotation is confirmed? What I would expect is a policy API. A user can define a policy that for example states:

```python
if len(newly_annotated_datapoints) > minimum_new_datapoints_before_retraining and model.score(newly_annotated_datapoints) < mode.score(test_set_of_previously_labeled_datapoints):
    model.fit(...)
```


In my case, the labeling backend also ever receives only a single task in the list of tasks sent to `LabelStudioMLBase.predict`. I don't know why. But the signature of `predict` in `LabelStudioMLBase` is

```
    @abstractmethod
    def predict(self, tasks, **kwargs):
        pass
```

So taskS. Plural. But my derived class only receives a one-element list of tasks.

**Comment by 1andDone** [comment]
@mbspng any resolution here? Currently dealing with a somewhat similar issue.

![image](https://github.com/HumanSignal/label-studio/assets/63823064/512ee81a-d456-46d3-ab00-c6b841d6db40)
